site/content/docs/main/upgrade-to-1.10.md

```diff
@@ -0,0 +1,91 @@
+---
+title: "Upgrading to Velero 1.10"
+layout: docs
+---
+
+## Prerequisites
+
+- Velero [v1.9.x][5] installed.
+
+If you're not yet running at least Velero v1.6, see the following:
+
+- [Upgrading to v1.5][1]
+- [Upgrading to v1.6][2]
+- [Upgrading to v1.7][3]
+- [Upgrading to v1.8][4]
+- [Upgrading to v1.9][5]
+
+Before upgrading, check the [Velero compatibility matrix](https://github.com/vmware-tanzu/velero#velero-compatabilty-matrix) to make sure your version of Kubernetes is supported by the new version of Velero.
+
+## Instructions
+
+**Caution:** From Velero v1.10, except for using restic to do file-system level backup and restore, kopia is also been integrated, so there would be a little bit of difference when upgrading to v1.10 from a version lower than v1.10.0.
+
+1. Install the Velero v1.10 command-line interface (CLI) by following the [instructions here][0].
+
+    Verify that you've properly installed it by running:
+
+    ```bash
+    velero version --client-only
+    ```
+
+    You should see the following output:
+
+    ```bash
+    Client:
+        Version: v1.10.0
+        Git commit: <git SHA>
+    ```
+
+1. Update the Velero custom resource definitions (CRDs) to include schema changes across all CRDs that are at the core of the new features in this release:
+
+    ```bash
+    velero install --crds-only --dry-run -o yaml | kubectl apply -f -
+    ```
+
+    **NOTE:** Since velero v1.10.0 only v1 CRD will be supported during installation, therefore, the v1.10.0 will only work on kubernetes version >= v1.16
+
+1. Update the container image and objects fields used by the Velero deployment and, optionally, the restic daemon set:
+
+    ```bash
+    # uploader_type value could be restic or kopia
+    kubectl get deploy -n velero -ojson \
+    | sed "s#\"image\"\: \"velero\/velero\:v[0-9]*.[0-9]*.[0-9]\"#\"image\"\: \"velero\/velero\:main\"#g" \
+    | sed "s#\"server\",#\"server\",\"--uploader-type=$uploader_type\",#g" \
+    | sed "s#default-volumes-to-restic#default-volumes-to-fs-backup#g" \
+    | sed "s#default-restic-prune-frequency#default-repo-maintain-frequency#g" \
+    | sed "s#restic-timeout#fs-backup-timeout#g" \
+    | kubectl apply -f -
+
+    # optional, if using the restic daemon set
+    dsjson=$(kubectl get ds -n velero -ojson)
+    kubectl delete ds -n velero --all --force --grace-period 0
+    echo $dsjson | sed "s#\"image\"\: \"velero\/velero\:v[0-9]*.[0-9]*.[0-9]\"#\"image\"\: \"velero\/velero\:v1.10.0\"#g" \
+    | sed "s#\"name\"\: \"restic\"#\"name\"\: \"node-agent\"#g" \
+    | sed "s#\[ \"restic\",#\[ \"node-agent\",#g" \
+    | kubectl apply -f -
+    ```
+
+1. Confirm that the deployment is up and running with the correct version by running:
+
+    ```bash
+    velero version
+    ```
+
+    You should see the following output:
+
+    ```bash
+    Client:
+        Version: v1.10.0
+        Git commit: <git SHA>
+
+    Server:
+        Version: v1.10.0
+    ```
+
```

Comment:
We also need to tell users that some left over resources need to be deleted manually:
`resticrepository` CRs
`velero-restic-credentials` secret

There is no problem to leave these unused resources there, but if users want to delete them, they need to do it manually through kubectl

Reply:
ok